src/coniql/caplugin.py

```diff
@@ -59,7 +58,6 @@ def _create_formatter(
         elif isinstance(value, (int, float)):
             # number
             formatter = ChannelFormatter.for_number(
-                display_form,
                 precision,
                 units,
             )
```

Comment:
Another thing that is strictly speaking not a fault of this PR, but I'm noting it here: The type hints for `for_number` appear to be incorrect - I can't see a way that `precision` and `units` could ever be `None`. The use of `getattr` will mean the values are `0` and `""` respectively, not `None`, so the `Optional` part of the type hint is unnecessary.

This also applies to `for_ndarray`.

Reply:
I agree that in this case they cannot be `None`, but in the \_\_init\_\_() of simplugin.py they could be as in the `ChannelDisplay` class both precision and units are `Optional`.